util/design/sparse-fsm-encode.py

```diff
@@ -238,15 +238,19 @@ def main():
 // This primitive is used to place a size-only constraint on the
 // flops in order to prevent FSM state encoding optimizations.
 logic [StateWidth-1:0] state_raw_q;
+logic state_valid;
+`PRIM_SPARSE_FSM_VALID_STATE(state_e, state_raw_q, state_valid)
+
 assign state_q = state_e'(state_raw_q);
-prim_flop #(
+prim_sparse_fsm_flop #(
   .Width(StateWidth),
   .ResetValue(StateWidth'(State0))
 ) u_state_regs (
   .clk_i,
   .rst_ni,
-  .d_i ( state_d     ),
-  .q_o ( state_raw_q )
+  .state_invalid_i ( state_valid ),
```

Comment:
Something looks wrong here! I think that `state_valid` is correctly named (it's the output of a call to `$cast`, which returns true on success). So we either need a `!` here or (possibly better) change the input port to `state_valid_i` and turn the logic around inside the primitive.

Reply:
o good catch!

Reply:
looks like we can remove this input now.